providers/common/include/prov/hybrid_pkey.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/evp.h>
+#include "internal/refcount.h"
+
+#define MAX_HYBRID_ALGS 2
+#define MAX_HYBRID_PARAMS 20
+
+typedef struct {
+    const unsigned int num_algs;
+    const struct {
+        const char *name;
+        size_t pubkey_length_bytes;
+        size_t bits;
+        size_t security_bits;
+        size_t shared_secret_bytes;
+        size_t ciphertext_bytes;
+    } alg[MAX_HYBRID_ALGS];
+} HYBRID_ALG_INFO;
+
+#define HYBRID_COMMON_FIELDS            \
+    OSSL_LIB_CTX *libctx;               \
```

Comment:
Is there a reason why HYBRID_COMMON_FIELDS could not be an array [2] also? (I assume its the higher level API)

Reply:
It is fundamentally not an array.  It contains information about the hybrid algorithm not about the sub algorithms.  The sub algorithms' information is in the constant `info` structure which is an array.